lgc/patch/PatchEntryPointMutate.cpp

```diff
@@ -284,538 +352,512 @@ bool PatchEntryPointMutate::isResourceNodeActive(const ResourceNode *node, bool
 }
 
 // =====================================================================================================================
-// Generates the type for the new entry-point based on already-collected info in LLVM context.
+// Generates the type for the new entry-point based on already-collected info.
+// This is what decides what SGPRs and VGPRs are passed to the shader at wave dispatch:
+//
+// * (For a GFX9+ merged shader or NGG primitive shader, the 8 system SGPRs at the start are not accounted for here.)
+// * The "user data" SGPRs, up to 32 (GFX9+ non-compute shader) or 16 (compute shader or <=GFX8). Many of the values
+//   here are pointers, but are passed as a single 32-bit register and then expanded to 64-bit in the shader code:
+//   - The "global information table", containing various descriptors such as the inter-shader rings
```

Comment:
In the initial development of LLPC,  we keep user data layout close to internal compiler's layout to speed up the development.
now, I think It is time to remove unnecessary field, include per-shader table and etc.

Reply:
Good idea for later commits...